libfoundation/src/foundation-string.cpp

```diff
@@ -4841,7 +4841,25 @@ static bool __MCStringExpandAt(MCStringRef self, uindex_t p_at, uindex_t p_count
 	// Fetch the capacity.
 	uindex_t t_capacity;
 	t_capacity = __MCStringGetCapacity(self);
-    
+
+	/* Overflow check.  A string's length needs to fit into an
+	 * index_t, because sometimes negative indices are used.  Ensure that the
+	 * length of the expanded string (including a null byte) fits. */
+	if (INDEX_MAX == p_count || INDEX_MAX - p_count - 1 < self->char_count)
```

Comment:
What's the reason for having to include the NUL byte - the NUL byte is not counted in the length of the string, so nowhere outside of libfoundation actually sees it.